crates/weaver_forge/codegen_examples/expected_codegen/http.rs

```diff
@@ -0,0 +1,139 @@
+
+
+/// HTTP request headers, `<key>` being the normalized HTTP Header name (lowercase), the value being the header values.
+///
+/// Notes:
+///   Instrumentations SHOULD require an explicit configuration of which headers are to be captured. Including all request headers can be a security risk - explicit configuration helps avoid leaking sensitive information.
+///   The `User-Agent` header is already captured in the `user_agent.original` attribute. Users MAY explicitly configure instrumentations to capture them even though it is not recommended.
+///   The attribute value MUST consist of either multiple header values as an array of strings or a single-item array containing a possibly comma-concatenated string, depending on the way the HTTP library provides access to headers.
+pub const HTTP_REQUEST_HEADER: AttributeKey<Vec<String>> = AttributeKey::new("http.request.header");
+
+
+/// HTTP request method.
+///
+/// Notes:
+///   HTTP request method value SHOULD be "known" to the instrumentation.
+///   By default, this convention defines "known" methods as the ones listed in [RFC9110](https://www.rfc-editor.org/rfc/rfc9110.html#name-methods)
+///   and the PATCH method defined in [RFC5789](https://www.rfc-editor.org/rfc/rfc5789.html).
+///   
+///   If the HTTP request method is not known to instrumentation, it MUST set the `http.request.method` attribute to `_OTHER`.
+///   
+///   If the HTTP instrumentation could end up converting valid HTTP request methods to `_OTHER`, then it MUST provide a way to override
+///   the list of known HTTP methods. If this override is done via environment variable, then the environment variable MUST be named
+///   OTEL_INSTRUMENTATION_HTTP_KNOWN_METHODS and support a comma-separated list of case-sensitive known HTTP methods
+///   (this list MUST be a full override of the default known method, it is not a list of known methods in addition to the defaults).
+///   
+///   HTTP method names are case-sensitive and `http.request.method` attribute value MUST match a known HTTP method name exactly.
+///   Instrumentations for specific web frameworks that consider HTTP methods to be case insensitive, SHOULD populate a canonical equivalent.
+///   Tracing instrumentations that do so, MUST also set `http.request.method_original` to the original value.
+pub const HTTP_REQUEST_METHOD: AttributeKey<HttpRequestMethod> = AttributeKey::new("http.request.method");
+
+
+/// HTTP request method.
+#[non_exhaustive]
+pub enum HttpRequestMethod {
+    /// CONNECT method.
+    Connect,
+    /// DELETE method.
+    Delete,
+    /// GET method.
+    Get,
+    /// HEAD method.
+    Head,
+    /// OPTIONS method.
+    Options,
+    /// PATCH method.
+    Patch,
+    /// POST method.
+    Post,
+    /// PUT method.
+    Put,
+    /// TRACE method.
+    Trace,
+    /// Any HTTP method that the instrumentation has no prior knowledge of.
+    Other,
```

Comment:
Those should have string values matching ones defined in the spec, i.e. `CONNECT`, `_OTHER` - is it possible in Rust?

Also, enums are notoriously difficult in other languages for their back-compat issues  (calling `Enum.NewValueAddedInVNext` but in runtime you deal with vOld of the enum vlass). I assume it's not an issue in Rust?

Reply:
Yes, there are ways to create this mapping, either with a macro or directly via the implementation of the Display trait. I will add that support.

To handle the evolution of enum definitions, I used the `#[non_exhaustive]` macro attribute, which indicates that a type or variant may have more fields or variants added in the future (more details [here](https://doc.rust-lang.org/reference/attributes/type_system.html)).

Reply:
The string values for the enums are now supported. See https://github.com/open-telemetry/weaver/pull/136/commits/9483d63349c12aaa53da99c3a6002f6eaab7aef9